include/oneapi/dpl/pstl/hetero/algorithm_impl_hetero.h

```diff
@@ -94,7 +113,7 @@ __pattern_walk2(_ExecutionPolicy&& __exec, _ForwardIterator1 __first1, _ForwardI
     auto __buf1 = __keep1(__first1, __last1);
 
     auto __keep2 = oneapi::dpl::__ranges::__get_sycl_range<__acc_mode2, _ForwardIterator2>();
-    auto __buf2 = __keep2(__first2, __first2 + __n);
+    auto __buf2 = __keep2(__first2, __first2 + get_buffer_size()(__first2, __n));
```

Comment:
I think we have to make buffer size request within `__ranges::__get_sycl_range` class.
That class takes into account a transformation pipeline over a passed  hetero iterator.

Reply:
ok, will do

Reply:
Fixed